### PR TITLE
Fix sbd print rate

### LINKED ISSUE #3409: `sbd_print_rate` is 1% in the haircut
When the haircut price applies, the virtual supply is:
`virtual_supply = current_supply + current_supply/9`

What is the `sbd_print_rate`?
https://github.com/steemit/steem/blob/a470bc283260e582b56897bd5c68107aa0289f18/libraries/chain/database.cpp#L3901-L3909

```percent_sbd = ( current_supply/9 * 10000 ) / ( current_supply + current_supply/9 )```

The problem with this formula is that all operations are made with integers. Meaning that the **percent_sbd will be 1000 only if the `current_supply` is divisible by 9. Otherwise it always be 999**.

If `percent_sbd == 999` then `sbd_print_rate=1%`.

### PR DESCRIPTION
**Pull request by mvandeberg**
Closes #3409

Local PR for #3410

> A previous fix was submitted in #3291
> However that solution creates new problems: The sbd print rate jumps from 5% to 0%.
> 
> In this PR I propose calculate percent_sbd by rounding the number instead of getting the floor (which is a intrinsic characteristic of integers). To do that I use the function described here.
> 
> This closes #3409 and #3184

### CODE REVIEW

File: libraries/chain/database.cpp [hunk 1]
```diff
@@ -3901,9 +3900,15 @@ void database::update_virtual_supply()
          auto percent_sbd = uint16_t( ( ( fc::uint128_t( ( dgp.current_sbd_supply * get_feed_history().current_median_history ).amount.value ) * STEEM_100_PERCENT )
```
Comment (sgerbino) [inline libraries/chain/database.cpp]: It looks like this statement would be unnecessary after hardfork 21 and should probably go in an `else` statement.

File: libraries/chain/database.cpp [hunk 2]
```diff
@@ -3898,12 +3897,22 @@ void database::update_virtual_supply()
...
+         uint16_t percent_sbd = 0;
+
+         if( has_hardfork( STEEM_HARDFORK_0_21 ) )
+         {
+            percent_sbd = uint16_t( ( ( fc::uint128_t( ( dgp.current_sbd_supply * get_feed_history().current_median_history ).amount.value ) * STEEM_100_PERCENT + dgp.virtual_supply.amount.value/2 )
+               / dgp.virtual_supply.amount.value ).to_uint64() );
+         }
+         else
+         {
+            percent_sbd = uint16_t( ( ( fc::uint128_t( ( dgp.current_sbd_supply * get_feed_history().current_median_history ).amount.value ) * STEEM_100_PERCENT )
```
Comment (eonwarped) [inline libraries/chain/database.cpp]: What about the HF20 behavior? It looks like it isn't happening here?

Comment (mvandeberg) [reply libraries/chain/database.cpp]: The else block is the HF 20 behavior. What about this looks wrong to you?

Comment (eonwarped) [reply libraries/chain/database.cpp]: The buffer of 5 for the stop print rate gap that was specific to HF20. And then HF19 had that parameter set to zero. Isn't this going to behave differently for the HF20 blocks?

Comment (joticajulian) [reply libraries/chain/database.cpp]: This buffer of 5 was only for the HF21. Not present in HF20 or HF19.

Comment (eonwarped) [reply libraries/chain/database.cpp]: Ah that's what I was missing. Thanks!

**Pull request merged**